File: app/src/main/java/com/pratik/moviesapp/viewmodels/MovieListViewModel.java
package com.pratik.moviesapp.viewmodels;

import android.arch.lifecycle.ViewModel;

import com.pratik.moviesapp.models.Movie;


public class MovieListViewModel extends ViewModel {

    private MovieRepository movieRepository;

    MovieListViewModel(MovieRepository repository) {
        super();
        this.movieRepository = repository;
    }

    public Movie syncPopularMovies() {
       return movieRepository.getPopularMovies();
    }

    public Movie syncTopRatedMovies() {
        return movieRepository.getTopRatedMovies();
    }
}

File: app/src/main/java/com/pratik/moviesapp/viewmodels/MovieListViewModel.kt
package com.pratik.moviesapp.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.pratik.moviesapp.controller.MovieCallBack
import com.pratik.moviesapp.controller.MovieController
import com.pratik.moviesapp.models.Movie
import retrofit2.Call

class MovieListViewModel(application: Application) : AndroidViewModel(application) {

    fun getPopularMovies(callBack: MovieCallBack?): Call<Movie?>? {
        return callBack?.let { MovieController.moviesPopular(it) }
    }

    fun getTopRatedMovies(callBack: MovieCallBack?): Call<Movie?>? {
        return callBack?.let { MovieController.getTopRatedMovies(it) }
    }

}